File: Studia/Projekt_Hurtowni/src/projekt_hurtowni/Telefon.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package projekt_hurtowni;
import java.io.*;
/**
 *
 * @author Bartek
 */
public class Telefon extends Produkt {
    String producent;
    String model;
    int szybkosc;
    int ekran;
    int bateria;
    int cena;
    public Telefon(String p, String m, int sz, int e, int b, int c)
    {
        producent = p;
        model = m;
        szybkosc = sz;
        ekran = e;
        bateria = b;
        cena = c;
    }
    @Override
    public void wyswietl_produkt()
    {
           System.out.print("*Telefon*    producent: " + producent + ", model: "
           + model + ",  szybkosc: " + szybkosc + ", ekran: " + ekran 
           + ", bateria: " + bateria + ", cena: " + cena);
    }
    /**
     *
     * @throws java.io.FileNotFoundException
     */
    @Override
    public void zapisz_produkt() throws FileNotFoundException
    {
        try (PrintWriter zapis = new PrintWriter(new FileOutputStream(new File("Telefon.txt"), true))) {
            zapis.append(producent + ":" + model + ":" + szybkosc + ":" + ekran + ":" + bateria + ":" + cena + ":");
            zapis.close();
        }
    }
}

File: Studia/Projekt_Hurtowni_c++/Telefon.cpp
#include "Classes/Telefon.hpp"
#include <fstream>

Telefon::Telefon(string p, string m, int sz, int e, int b, int c) : Produkt(p,m,c)
{
	szybkosc = sz;
	ekran = e;
	bateria = b;
}

void Telefon::wyswietl_produkt()
{
	cout<<"*Telefon*	Producent: "<<producent<<", Model: "<<model<<", szybkosc: "<<szybkosc<<", ekran: "<<ekran<<", wytrzymalosc baterii: "<<bateria<<
	", cena:"<<cena;
}

void Telefon::zapisz_produkt()
{
	fstream plik;
	plik.open("Telefony.txt", ios::app);
	if(plik.good())
	{
		plik<<producent<<" "<<model<<" "<<szybkosc<<" "<<ekran<<" "<<bateria<<" "<<cena<<" ";
	}
	plik.close();
}

